runtime/type_conversion.cu

```diff
@@ -0,0 +1,32 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+
+// This should be specialized for each combination of types that cannot be
+// automatically cast by static_cast
+template <typename TO, typename FROM>
+__device__ __inline__ TO castFloating(FROM x) {
```

Comment:
TODO: we could support vectorized casts here by adding a `vec_size` template arg then specializing to the usual set of vectorization widths and using `__half22float2` and friends.